src/containers/blocks.jsx

```diff
@@ -167,6 +191,12 @@ class Blocks extends React.Component {
             this.workspace.resize();
         }
     }
+    onExtensionWasAdded (blocksInfo) {
+        this.ScratchBlocks.defineBlocksWithJsonArray(blocksInfo.map(blockInfo => blockInfo.json));
+        const dynamicBlocksXML = this.props.vm.runtime.getBlocksXML();
+        const toolboxXML = makeToolboxXML(dynamicBlocksXML);
+        this.props.sendToolboxUpdate(toolboxXML);
```

Comment:
Just a question, not a change request: @cwillisf is it possible to bind the vm directly to scratch-blocks to handle toolbox updates? Does the GUI have to be involved with that part beyond hooking the vm up with scratch-blocks, similar to the way the blocks events are connected directly with the vm? @rschamp do you think there are any consequences of doing one way vs. the other?

Reply:
I think we should keep scratch-blocks specific logic to a minimum in the VM, so that the VM isn't concerned with UI.  I also think it's a good design to not bind the blocks to the VM given how much we mutate scratch-blocks currently.